tests/cupy_tests/core_tests/test_raw.py

```diff
@@ -665,6 +740,64 @@ def test_context_switch_RawModule4(self):
         x1, x2, y = self._helper(ker, cupy.float32)
         assert cupy.allclose(y, x1 / (x2 + 1.0))
 
+    @testing.multi_gpu(2)
+    def test_context_switch_RawModule5(self):
+        # run test_template_specialization() on another device
+        # in this test, re-compiling happens at get_function()
+        if self.backend == 'nvcc':
+            self.skipTest('nvcc does not support template specialization')
+
+        # compile code
+        specializations = ['my_sqrt<unsigned int>']
+        mod = cupy.RawModule(code=test_cxx_template, options=('--std=c++11',),
+                             specializations=specializations)
+
+        # switch device
+        cupy.cuda.runtime.setDevice(1)
```

Comment:
Could you use `with cupy.cuda.Device(1)` to switch the current device? It automatically restores the current device at the end of the context.